webdav/lock.go

```diff
@@ -93,6 +93,19 @@ type LockSystem interface {
 	Unlock(now time.Time, token string) error
 }
 
+// LockDeleter extends a LockSystem to support deleting locks by file. This is
+// useful when the LockSystem is decoupled from the FileSystem, particularly
+// because some FileSystem operations impact the LockSystem (e.g. deleting a
+// file).
+type LockDeleter interface {
```

Comment:
This is the only way I could think of cleaning up locks based on resource name without breaking backwards compatibility with existing implementations of LockSystem. If someone has a better approach in mind, I'm willing to give it a try.